extensions/wikia/Chat2/ChatConfig.class.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+class ChatConfig {
+
+	// constants with config file sections
+	const CHAT_DEVBOX_ENV = 'dev';
+	const CHAT_PREVIEW_ENV = 'preview';
+	const CHAT_VERIFY_ENV = 'verify';
+	const CHAT_PRODUCTION_ENV = 'prod';
+
+	const SERVER_TYPE_MAIN = 'Main';
+	const SERVER_TYPE_API = 'Api';
+
+	const VAR_SERVER_BASKET = 'wgChatServersBasket';
+	const VAR_OPERATION_MODE = 'wgChatOperationMode';
+	const CENTRAL_WIKI_ID = 177;
+
+	private static $configData = [ ];
+
+	public static function getMainServer() {
+		return self::getServer( self::SERVER_TYPE_MAIN );
+	}
+
+	public static function getApiServer() {
+		return self::getServer( self::SERVER_TYPE_API );
+	}
+
+	public static function getMainServersList() {
+		return self::getServersList( self::SERVER_TYPE_MAIN );
+	}
+
+	private static function getServer( $type ) {
+		global $wgCityId;
+
+		$servers = self::getServersList( $type );
+		$index = $wgCityId % count( $servers );
+
+		list( $host, $port ) = explode( ':', $servers[$index], 2 );
+
+		return [
+			'host' => $host,
+			'port' => $port,
+			'serverId' => $index + 1,
+		];
+	}
+
+	public static function getSecretToken() {
+		return ChatConfig::getConfigValue( 'ChatCommunicationToken' );
+	}
+
+	private static function getServersList( $type ) {
+		$servers = self::getConfigValue( $type . 'ChatServers' );
+
+		return $servers[self::getServerBasket()];
+	}
+
+	public static function getPublicHost() {
+		return self::getConfigValue( 'ChatHost' );
+	}
+
+	/**
+	 * Get a configuration value. Returns false when not found.
+	 *
+	 * @param string $key Key
+	 * @return mixed|false
+	 */
+	private static function getConfigValue( $key ) {
+		global $wgWikiaEnvironment;
+
+		if ( empty( self::$configData ) ) {
+			$configDir = getenv( 'WIKIA_CONFIG_ROOT' );
+			$configFile = $configDir . '/ChatConfig.json';
```

Comment:
Could we just do `$configFile = getenv( 'WIKIA_CONFIG_ROOT' ) . '/ChatConfig.json';` ?